apps/ejabberd/src/mod_mam_muc.erl

```diff
@@ -260,7 +225,7 @@ filter_room_packet(Packet, EventData) ->
         FromJID :: ejabberd:jid(), RoomJID :: ejabberd:jid(),
         Role :: mod_muc:role(), Affiliation :: mod_muc:affiliation()) -> packet().
 archive_room_packet(Packet, FromNick, FromJID=#jid{}, RoomJID=#jid{}, Role, Affiliation) ->
-    Host = server_host(RoomJID),
+    {_, Host} = gen_mod:get_module_and_host_by_subhost(RoomJID#jid.lserver),
```

Comment:
I see there is many cases where only the `Host` is needed. Maybe it would be worth to create a function which returns just `Host`? `gen_mod:get_host_by_subhost`?